synapseclient/client.py

```diff
@@ -3853,47 +3853,104 @@ def _find_teams_for_principal(self, principal_id: str) -> typing.Iterator[Team]:
         for result in self._GET_paginated(f"/user/{principal_id}/team"):
             yield Team(**result)
 
-    @tracer.start_as_current_span("Synapse::getTeam")
-    def getTeam(self, id):
+    def create_team(
+        self,
+        name: str,
+        description: str = None,
+        icon: str = None,
+        opentelemetry_context=None,
+    ) -> Team:
+        """
+        Creates a new team.
+
+        Arguments:
+            name: The name of the team to create.
+            description: A description of the team.
+            icon: The FileHandleID of the icon to be used for the team.
+            opentelemetry_context: OpenTelemetry context to propogate to this function to use for tracing.
+                                    Used in cases where concurrent operations need to be linked to parent spans.
+
+        Returns:
+            An object of type [synapseclient.team.Team][]
+        """
+        with tracer.start_as_current_span(
+            "Synapse::create_team", context=opentelemetry_context
+        ):
+            request_body = {"name": name, "description": description, "icon": icon}
+            return Team(
+                **self.restPOST(
+                    "/team",
+                    json.dumps(request_body),
+                )
+            )
+
+    def delete_team(self, id: int, opentelemetry_context=None) -> None:
```

Comment:
Same thing as above, you can delete a team by doing 

```
syn.delete(Team(...))
```

Reply:
As it stands, `syn.delete` does not work by simply passing it the new model, and you cannot delete a team by just passing its `id` to `syn.delete` as you can with entities that have a `syn123` identifier. So then to leverage the existing function I would have to convert the new model back to the old one which seems counterintuitive if our goal is to eventually transition to only using the new ones.